examples/FireboltPrivacy/FireboltPrivacyImplementation.cpp

```
@@ -0,0 +1,189 @@
+            }
+            ~Config() override = default;
+        public:
+            Core::JSON::EnumType<Exchange::IFireboltPrivacy::StorageLocation> StorageLocation;
+        };
+
+
+    public:
+        using Notifications = std::vector<Exchange::IFireboltPrivacy::INotification*>;
+        FireboltPrivacyImplementation(const FireboltPrivacyImplementation&) = delete;
+        FireboltPrivacyImplementation(FireboltPrivacyImplementation&&) = delete;
+        FireboltPrivacyImplementation& operator= (const FireboltPrivacyImplementation&) = delete;
+        FireboltPrivacyImplementation& operator= (FireboltPrivacyImplementation&&) = delete;
+        FireboltPrivacyImplementation()
+            : _service(nullptr)
+            , _adminLock()
+            , _allowResumePoints(false)
+            , _notifications()
+            , _inMemory(false)
+            , _localStore(nullptr)
+        {
+        }
+        ~FireboltPrivacyImplementation() override = default;
+
+        Core::hresult Register(Exchange::IFireboltPrivacy::INotification* notification) override
+        {
+            ASSERT(notification != nullptr);
+            ASSERT(std::find(_notifications.begin(), _notifications.end(), notification) == _notifications.end());
+            _adminLock.Lock();
+            notification->AddRef();
+            _notifications.push_back(notification);
+            _adminLock.Unlock();
+
+            return Core::ERROR_NONE;
+        }
+
+        Core::hresult Unregister(Exchange::IFireboltPrivacy::INotification* notification) override
+        {
+            ASSERT(notification != nullptr);
+            _adminLock.Lock();
+            auto item = std::find(_notifications.begin(), _notifications.end(), notification);
+            if(item != _notifications.end()){
+                notification->Release();
+                _notifications.erase(item);
+            }
+            _adminLock.Unlock();
+
+            return Core::ERROR_NONE;
```

Please report an error if you could not unregister it, also ASSERT on it if the notification could not be found. 
We promote defensive coding these ASSERTS help in these defensive coding.